five/intid/keyreference.py

```diff
@@ -87,6 +89,13 @@ def __init__(self, wrapped_obj):
         self.oid = self.object._p_oid
         self.dbname = connection.db().database_name
 
+    def __setstate__(self, state):
+        for key in ('root_oid', 'oid'):
+            value = state.get(key)
+            if isinstance(value, six.text_type):
+                state[key] = value.encode('utf-8')
+        self.__dict__.update(state)
```

Comment:
I'm not sure how often this code will run but maybe we should do `self.__dict__.update(state)` only if something actually changed?